app/screens/channel/channel_base.js

```diff
@@ -123,6 +126,7 @@ export default class ChannelBase extends PureComponent {
 
         if (nextProps.currentChannelId !== this.props.currentChannelId &&
             nextProps.currentTeamId === this.props.currentTeamId) {
+            nextProps.actions.getChannelStats(nextProps.currentChannelId);
```

Comment:
you should move this its own if statement as this will not trigger when you switch teams.

```js
if (nextProps.currentChannelId !== this.props.currentChannelId) {
    this.props.actions.getChannelStats(nextProps.currentChannelId);
}
```

Reply:
Done! Thanks :)